### PR TITLE
Add "resolver = 2" option to workspace manifest examples

### PR DESCRIPTION
**Pull request by mousetail**
It seems like `resolver = 2` is just a basic requirement to build any package with dependencies. To make the examples work it's needed. Apparently there is work to make it default (https://github.com/rust-lang/rust/issues/90148) but right now it's required.

### What does this PR try to resolve?

The examples in the documentation about workspaces don't work in the latest edition of rust. This would have saved me many hours of debugging.

Ideally there would also be a description of what the option actually does but I don't really understand that myself.

### How should we test and review this PR?

It's a documentation change. Check if you think the information is useful.

### Additional information

Relevant links:
- https://github.com/rust-lang/rust/issues/90148
- https://github.com/rust-lang/cargo/issues/9996
- https://doc.rust-lang.org/nightly/edition-guide/rust-2021/default-cargo-resolver.html#details

### CODE REVIEW

**Comment by epage** [comment]
FYI I updated the PR description to more closely match what this does.  When I read it, I assumed it was going to apply to all manifest examples, even if the edition is set.

File: src/doc/src/reference/workspaces.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ the workspace:
 [workspace]
 members = ["member1", "path/to/member2", "crates/*"]
 exclude = ["crates/foo", "path/to/other"]
+resolver = "2"
```
[the remark this answers is not in the document]
Comment (mousetail) [reply src/doc/src/reference/workspaces.md:45]: Good feedback. Unfortunately I don't really know enough about how the resolver works to properly write that paragraph.

[the remark this answers is not in the document]
Comment (epage) [reply src/doc/src/reference/workspaces.md:45]: I think you could scavenge some content from https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions and link to it for more details.

Browsing to that, another variant of the re-structure I mentioned would be to model it after [the break down of the full manifest format](https://doc.rust-lang.org/cargo/reference/manifest.html) where we have a ToC that represents the structure of `[workspace]` table

[the remark this answers is not in the document]
Comment (mousetail) [reply src/doc/src/reference/workspaces.md:45]: Ok I added a short paragraph, though I'm not 100% sure if what I wrote is correct

**Comment by weihanglo** [comment]
Ping @mousetail. Just checking in to see if you are still interested in working on this, or if you had any questions.

By the way, [workspace.md](https://github.com/rust-lang/cargo/blob/master/src/doc/src/reference/workspaces.md) has got a content re-org recently. You may need to resolve some conflicts before proceeding.

File: src/doc/src/reference/workspaces.md [hunk 2]
```diff
@@ -101,6 +102,15 @@ workspace. This can be useful if some path dependencies aren't desired to be
 in the workspace at all, or using a glob pattern and you want to remove a
 directory.
 
+The [`resolver`] key effects how a package that is depended on in multiple
+ways with different features are resolved. Some libraries depend on it being
+set to 2.
+
+An empty `[workspace]` table can be used with a `[package]` to conveniently
+create a workspace with the package and all of its path dependencies.
```
Comment (epage) [inline src/doc/src/reference/workspaces.md:110]: This doesn't quite fit within the current section

Maybe we should move this into the "virtual workspace" section and only update that example?

Comment (mousetail) [reply src/doc/src/reference/workspaces.md]: I removed this paragraph, seems like this concept is explained well enough in the virtual workspace section

Comment (epage) [reply src/doc/src/reference/workspaces.md]: The focus of my concern wasn't so much the duplication of information about virtual workspaces but that we are in the section about members and exclude keys and introducing the resolver key as if it belongs in that section.  We then include the resolver key in each further section.  I get the point of including the resolver key in each subsequent section, to try to remind people to  use it, but it doesn't quite fit seeing an unexplained key and one that is only relevant when its a virtual workspace.  This is why I was recommending to make this entire PR only touch the virtual workspace section.

This PR does not have to do all of the lifting for helping users.  We also have https://github.com/rust-lang/cargo/pull/10910.

File: src/doc/src/reference/workspaces.md [hunk 3]
```diff
@@ -101,6 +107,8 @@ workspace. This can be useful if some path dependencies aren't desired to be
 in the workspace at all, or using a glob pattern and you want to remove a
 directory.
 
+### Workspace selection
+
```
Comment (epage) [inline src/doc/src/reference/workspaces.md:111]: Please remove

File: src/doc/src/reference/workspaces.md [hunk 4]
```diff
@@ -128,6 +136,7 @@ used:
 [workspace]
 members = ["path/to/member1", "path/to/member2", "path/to/member3/*"]
 default-members = ["path/to/member2", "path/to/member3/foo"]
+resolver = "2"
```
Comment (epage) [inline src/doc/src/reference/workspaces.md:139]: Please remove

File: src/doc/src/reference/workspaces.md [hunk 5]
```diff
@@ -225,6 +234,7 @@ configuration in `Cargo.toml`. For example:
 ```toml
 [workspace]
 members = ["member1", "member2"]
+resolver = "2"
```
Comment (epage) [inline src/doc/src/reference/workspaces.md:237]: Please remove

File: src/doc/src/reference/workspaces.md [hunk 6]
```diff
@@ -78,6 +83,7 @@ version = "0.1.0"    # the current version, obeying semver
 authors = ["Alice <a@example.com>", "Bob <b@example.com>"]
 ```
 
+
```
Comment (epage) [inline src/doc/src/reference/workspaces.md:86]: Please remove

File: src/doc/src/reference/workspaces.md [hunk 7]
```diff
@@ -68,8 +68,13 @@ you want to keep all the packages organized in separate directories.
 # [PROJECT_DIR]/Cargo.toml
 [workspace]
 members = ["hello_world"]
+resolver = "2"
 ```
 
+The [`resolver`] key here effects how a package that is depended on in multiple
+ways with different features are resolved. Some libraries depend on it being
+set to 2.
```
Comment (epage) [inline src/doc/src/reference/workspaces.md:76]: I'd recommend shifting the focus to fit within this section.  We delegate explanation of the field to the resolver page.

Something like

```md
Without a root package to fall back on, the [`resolver`] field should be set in the `[workspace]` table.
```
(at first I was worried this was too strong of a recommendation but it is roughly what the `resolver` page says)

**Comment by weihanglo** [comment]
Ping @mousetail. Just checking in to see if you are still interested in working on this doc improvement. Feel free to ask any question here or on Zuilp.

**Comment by ehuss** [comment]
I'm going to close due to inactivity. I think it might be good to more clearly recommend setting the resolver field in a workspace, but I'm not sure this PR quite hit the right balance of focus. I'd also mention #10112 might be a stronger indicator than assuming people read the documentation.

Thanks!

**Pull request closed without merging**